### ISSUE TITLE
'Vertical viewport offset' leads to display problems

### ISSUE THREAD

**Issue by HilliBilli13** [Stellarium/stellarium#904]
See also: #1639488

When the vertical viewport offset ≠ 0°, there are display problems with
a) Flip scene vertically
b) Image sensor frame / Centering an object with mouse double-click
c) Ocular view

### Expected Behaviour
A centered object should stay centered.

### Actual Behaviour
a) If you have a centered object and you do a 'flip scene vertically', then the object is no longer centered.
If you try to select and center another object this will also fail - in the middle of the display you see an object depending on the 'vertical viewport offset'.
------	
b) If you have selected  'Image sensor frame' on a centered object and you zoom in or out then the centered object moves out of the image sensor frame.
	
!! It works correctly if you center the object by SPACE and not by mouse double-click!!!
	
If it works correctly the zoom (only with key: page up/down) first zooms (and moves the object out of the rectangle) and then it centers to the selected object, this means it moves it into the center within one second.
But: With scroll wheel the zoom works perfectly!
	
Zooming out extremely: The centered object moves out of the rectangle!?
------		
c) If you center an object and then switch-on the 'ocular view', then the centered object will not be in the center of the ocular view - if in the telescope options 'Vertical flip' is selected or if you do a 'flip scene vertically'.
This is probably the same problem as for point a).
	
### Steps to reproduce	
	1. Change 'vertical viewport offset' to -25%
	2. Select and center an object
	3. Point a) Hit the 'Flip scene vertically' [Ctrl+Shift+'V']
	              Select another object and try to center it.
	Point b) Activate 'Image sensor frame' and zoom in or out.
	             - To center the object with SPACE key is different to center with mouse double-click.
	             - Zooming with page up/down key is different to zooming with scroll wheel.
	Point c)  Select 'Ocular view' [Ctrl+O] and hit the 'Flip scene vertically' [Ctrl+Shift+'V'] 
	
### System
* Stellarium version: 0.19.3
* Operating system: Win10 64 bit
* Graphics Card: AMD Radeon R7 200 Series

**Comment by gzotti** [comment]
Yes, there are a few minor issues around vertical offset. It has been constructed for wider-angle views that show horizon and sky, but more of the sky. You can use Cylindrical, Mercator or Stereographic mode and show more than half of the sky with that. Such scenes are typically never flipped. I prefer mouse wheel, so any other zoom method may have seen less attention. Some of the current behaviour was indeed required for a practical application context. 
A quick way out could be to disable vertical offset whenever the Oculars plugin is active. Telescope users typically won't need it. People who frame wide-angle shots probably could also live with non-offset screen geometry.
However, the part of Stellarium that includes all those various methods of moving around the sky has indeed become super-complicated over the years and deserves a more systematic re-write that also could include full touch screen functionality, and solve #754. Who dares?

**Comment by alex-w** [comment]
Vertical flip are working wrong, when vertical offset is defined.

**Comment by lijozech-12** [comment]
Hi, I Know C++ STL libraries, and OOPs. I Didn't have that much experience in open-source contributions. Can anyone guide me to solve this issue?

**Comment by alex-w** [comment]
This is a good task for the community to participate in the contribution into Stellarium. Who wants to help us?
